contracts/pcv/utils/ERC20Allocator.sol

```diff
@@ -0,0 +1,454 @@
+    mapping(address => PSMInfo) public allPSMs;
+
+    /// @notice map the pcv deposit address to a peg stability module
+    mapping(address => address) public pcvDepositToPSM;
+
+    /// @notice ERC20 Allocator constructor
+    /// @param _core Volt Core for reference
+    /// @param _maxRateLimitPerSecond maximum rate limit per second that governance can set
+    /// @param _rateLimitPerSecond starting rate limit per second
+    /// @param _bufferCap cap on buffer size for this rate limited instance
+    constructor(
+        address _core,
+        uint256 _maxRateLimitPerSecond,
+        uint128 _rateLimitPerSecond,
+        uint128 _bufferCap
+    )
+        CoreRef(_core)
+        RateLimitedV2(_maxRateLimitPerSecond, _rateLimitPerSecond, _bufferCap)
+    {}
+
+    /// ----------- Governor Only API -----------
+
+    /// @notice create a new deposit
+    /// @param psm Peg Stability Module for this deposit
+    /// @param targetBalance target amount of tokens for the PSM to hold
+    /// @param decimalsNormalizer decimal normalizer to ensure buffer is depleted and replenished properly
+    function createDeposit(
+        address psm,
+        uint248 targetBalance,
+        int8 decimalsNormalizer
+    ) external override onlyGovernor {
+        address token = PCVDeposit(psm).balanceReportedIn();
+
+        require(
+            allPSMs[psm].token == address(0),
+            "ERC20Allocator: cannot overwrite existing deposit"
+        );
+
+        PSMInfo memory newDeposit = PSMInfo({
+            token: token,
+            targetBalance: targetBalance,
+            decimalsNormalizer: decimalsNormalizer
+        });
+        allPSMs[psm] = newDeposit;
+
+        emit DepositCreated(psm, token, targetBalance, decimalsNormalizer);
+    }
+
+    /// @notice edit an existing deposit
+    /// @param psm Peg Stability Module for this deposit
+    /// @param targetBalance target amount of tokens for the PSM to hold
+    /// @param decimalsNormalizer decimal normalizer to ensure buffer is depleted and replenished properly
+    /// cannot manually change the underlying token, as this is pulled from the PSM
+    /// underlying token is immutable in both pcv deposit and
+    function editDeposit(
+        address psm,
+        uint248 targetBalance,
+        int8 decimalsNormalizer
+    ) external override onlyGovernor {
+        address token = PCVDeposit(psm).balanceReportedIn();
+        address storedToken = allPSMs[psm].token;
+        require(
+            storedToken != address(0),
+            "ERC20Allocator: cannot edit non-existent deposit"
+        );
+        /// assert pcv deposit and psm share same denomination
+        require(token == storedToken, "ERC20Allocator: psm changed underlying");
+
+        PSMInfo memory depositToEdit = PSMInfo({
+            token: token,
+            targetBalance: targetBalance,
+            decimalsNormalizer: decimalsNormalizer
+        });
```

Comment:
save an SSTORE here by only writing targetBalance and decimalsNormalizer to storage, token is unchanged.